### PR TITLE
[ISSUE-952][FEATURE] support register shuffle task in map partition mode

### PR DESCRIPTION
**Pull request by RexXiong**
see issue link #952 

# [BUG]/[FEATURE] title

### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

File: common/src/main/scala/org/apache/celeborn/common/util/Utils.scala [hunk 1]
```diff
@@ -606,18 +606,33 @@ object Utils extends Logging {
...
+
   def splitShuffleKey(shuffleKey: String): (String, Int) = {
     val splits = shuffleKey.split("-")
     val appId = splits.dropRight(1).mkString("-")
     val shuffleId = splits.last.toInt
     (appId, shuffleId)
   }
 
-  def splitPartitionLocationUniqueId(uniqueId: String): (Int, Int) = {
+  def splitPartitionLocationUniqueId(uniqueId: String): (Int, Int, Int) = {
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/util/Utils.scala:623]: let's document the return values

File: common/src/main/scala/org/apache/celeborn/common/util/Utils.scala [hunk 2]
```diff
@@ -606,18 +606,33 @@ object Utils extends Logging {
...
+    splits match {
+      case r if r.size == 3 =>
+        val partitionId = splits(0).toInt
+        val attemptId = splits(1).toInt
+        val epoch = splits.last.toInt
+        (partitionId, attemptId, epoch)
+      case _ =>
+        val partitionId = splits(0).toInt
+        val epoch = splits.last.toInt
+        (partitionId, 0, epoch)
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/util/Utils.scala:634]:
```suggestion
      case Array(partitionId, attemptId, epoch) => (partitionId, attemptId, epoch)
      case Array(partitionId, epoch) => (partitionId, 0, epoch)
      case unsupported => // balabala
```

File: common/src/test/scala/org/apache/celeborn/common/util/UtilsSuite.scala [hunk 3]
```diff
@@ -80,7 +80,8 @@ class UtilsSuite extends RssFunSuite {
   }
 
   test("splitPartitionLocationUniqueId") {
-    assert((1, 1).equals(Utils.splitPartitionLocationUniqueId("1-1")))
+    assert((1, 0, 1).equals(Utils.splitPartitionLocationUniqueId("1-1")))
+    assert((1, 1, 1).equals(Utils.splitPartitionLocationUniqueId("1-1-1")))
```
Comment (pan3793) [inline common/src/test/scala/org/apache/celeborn/common/util/UtilsSuite.scala:84]:
```suggestion
    assert(Utils.splitPartitionLocationUniqueId("1-1") === (1, 0, 1))
    assert(Utils.splitPartitionLocationUniqueId("1-1-1") === (1, 1, 1))
```

File: common/src/main/scala/org/apache/celeborn/common/util/Utils.scala (hunk 2 above)
Comment (RexXiong) [reply common/src/main/scala/org/apache/celeborn/common/util/Utils.scala:634]: 👍 Thanks. Very good advice.

**Comment by waitinfuture** [comment]
As we discussed offline, we'd better encode mapId and attemptId into partitionId instead of introducing group.

**Pull request closed without merging**